src/test/scala/treadle/RandomizeCircuit.scala

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 The Regents of the University of California (Regents)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package treadle
+
+import firrtl.stage.FirrtlSourceAnnotation
+import logger.{LazyLogging, LogLevel, Logger}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import treadle.chronometry.Timer
+
+import scala.util.Random
+
+class RandomizeCircuit extends AnyFreeSpec with Matchers with LazyLogging {
+  val t = new Timer()
+
+  def method1: Unit = {
+    Random.setSeed(System.currentTimeMillis())
+    t("method1") {
+      val a = Array.tabulate(1000) { i =>
+        Random.nextInt()
+      }
+      val sum = a.sum
+      println(s"method1 sum $sum")
+    }
+  }
+
+  def method2: Unit = {
+    t("method2") {
+      val a = Array.tabulate(1000) { i =>
+        Random.setSeed(i)
+        Random.nextInt()
+      }
+      val sum = a.sum
+      println(s"method2 $sum")
+    }
+  }
+
+  "random numbers should behave properly" in {
+    method1
+    method2
+    method1
+    method2
+    t.clear()
+    for (i <- 0 to 10) {
+      method1
+      method2
+    }
+    println(t.report())
```

Comment:
can you turn the check into assertions instead of printing to the command line?

Reply:
changed to test of quality here.